src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -158,6 +222,140 @@ async def generate_lockfile(
         ),
     )
 
+
+@rule
+async def generate_updated_lockfile(
+    update_req: UpdatePythonLockfileRequest,
+    generate_lockfiles_subsystem: GenerateLockfilesSubsystem,
+    pex_lock_subsystem: PexLockSubsystem,
+    python_setup: PythonSetup,
+) -> ProcessResult:
+    req = update_req.gen_req
+    resolve_config = await Get(ResolvePexConfig, ResolvePexConfigRequest(req.resolve_name))
+
+    for project in pex_lock_subsystem.project:
+        if PipRequirement.parse(project).project_name != project:
+            raise ValueError(
+                f"project {project} is not a bare name; specifier, markers must be specified in Pants targets"
+            )
+
+    # Assuming that any lockfile we are trying to update has a destination
+    # that is a file and not an embedded read only resource.  See
+    # pex_requirements.py
+    maybe_lockfile = await Get(
+        Digest,
+        PathGlobs(
+            [req.lockfile_dest],
+            glob_match_error_behavior=GlobMatchErrorBehavior.ignore,
+        ),
+    )
+    if maybe_lockfile.serialized_bytes_length == 0:
+        raise ValueError(
+            f"Empty or non-existent lockfile at {req.lockfile_dest}. Unable to update in place"
+        )
+
+    original_loaded_lockfile = await Get(
+        LoadedLockfile,
+        LoadedLockfileRequest(
+            Lockfile(
+                url=req.lockfile_dest,
+                url_description_of_origin=f"existing lockfile from {req.resolve_name}",
+                resolve_name=req.resolve_name,
+            )
+        ),
+    )
+
+    if not (
+        original_loaded_lockfile.metadata
+        and isinstance(original_loaded_lockfile.metadata, (PythonLockfileMetadataV3))
+    ):
+        raise ValueError(
+            f"Pants metadata for lockfile at {req.lockfile_dest} is in old <V3 format. Unable to update in place."
+        )
+    elif set(resolve_config.no_binary) != original_loaded_lockfile.metadata.no_binary:
+        raise ValueError(
+            f"Resolve Config no_binary {set(resolve_config.no_binary)} does not match {original_loaded_lockfile.metadata.no_binary} in last lockfile, can not update in place"
```

Comment:
Do we know the name of the resolve here, to augment this error message and tell the user which resolve might be the issue?

```suggestion
            f"In resolve {name}, configured no_binary {set(resolve_config.no_binary)} does not match {original_loaded_lockfile.metadata.no_binary} in last lockfile, can not update in place"
```